api/metadata.go

```diff
@@ -19,19 +18,18 @@ func (api *DatasetAPI) getMetadata(w http.ResponseWriter, r *http.Request) {
 	datasetID := vars["id"]
 	edition := vars["edition"]
 	version := vars["version"]
-	logData := log.Data{"dataset_id": datasetID, "edition": edition, "version": version}
 	auditParams := common.Params{"dataset_id": datasetID, "edition": edition, "version": version}
+	logData := audit.ToLogData(auditParams)
 
 	if auditErr := api.auditor.Record(ctx, getMetadataAction, audit.Attempted, auditParams); auditErr != nil {
-		auditActionFailure(ctx, getMetadataAction, audit.Attempted, auditErr, logData)
 		handleMetadataErr(w, auditErr)
 		return
 	}
 
 	b, err := func() ([]byte, error) {
 		datasetDoc, err := api.dataStore.Backend.GetDataset(datasetID)
 		if err != nil {
-			logError(ctx, errors.WithMessage(err, "getMetadata endpoint: get datastore.getDataset returned an error"), logData)
+			audit.LogError(ctx, errors.WithMessage(err, "getMetadata endpoint: get datastore.getDataset returned an error"), logData)
 			return nil, err
```

Comment:
Im not convinced these should default to audit logs; if inline function errors, the encapsulating function getMetadata will act on the error and call auditor Record function; which will in turn create an audit log of the action and the error result. So by making this log line an audit one we are polluting are audit logs, remember the reason for auditing is to audit actions not to use them for debugging?

Instead think about using log.ErrorCtx(...) instead.

The above can be applied to all logs within the inline functions.